### ISSUE TITLE
Indicate prominently whether current tab is already stashed

### ISSUE THREAD

**Issue by alphapapa** [josh-berry/tab-stash#406]
### Problem Statement

I use Tab Stash to help keep track of, e.g. YouTube videos of concerts and performances that I'm listening to and learning about.  The only remaining problem is that, if I choose a video to listen to from my Tab Stash, and then later need to re-stash it, I may not remember whether it already exists in the Stash--if it does, I could just close the tab and not lose it; but if it doesn't, I need to stash it again or else lose it.  

So before I can close the tab, I have to manually look through the stash to see if it already exists.  If I do come upon it, it's helpfully highlighted in blue, but I have to come across its location in order to see that, which means either manually expanding relevant folders and scrolling long lists, or typing some words from the title or URL and hoping to see it in the results.

### Preferred Solution(s)

* I would like it if, when I open any of the 3 Tab Stash UIs (tab, sidebar, drop-down menu), the item showing the current tab would indicate whether it already exists as a stashed tab.
* It would also be helpful if the toolbar button indicated this, e.g. by being in a different color or using a different icon.  This way I would know immediately whether I can just close the tab or need to stash it.

### Additional Context

Tab Stash is, somehow, more than the sum of its parts.  It's the only extension that has allowed me to easily keep the number of open tabs to a minimum without feeling like I would lose some information that I wanted to recall later.  Thank you for your work on it.  If this feature were added, I think it would be the final piece of the puzzle, for me.

Also, https://github.com/josh-berry/tab-stash/issues/232 mentions this idea as:

> Some visual indication for open tabs that have already been stashed. Probably also add the stash older to the tab tooltip. Could be behind an option that defaulted to off.

However, as this feature request template mentions, these ideas should be posted separately; especially, I think this one is significant and worthy of its own issue.

<hr>
<em>Readers: If you are also interested in seeing this feature be developed, please vote for it by giving the ORIGINAL POST a thumbs-up using the :smiley: button.  You are welcome to leave comments and discuss the feature request, but "Me too!" comments are not counted by the voting system.</em>

**Comment by josh-berry** [comment]
Hi, there are a couple ways you can tell if a tab is already stashed or not:

1. Without even opening Tab Stash, Firefox will helpfully color in the bookmark icon (which looks like a star) in the address bar if the URL is stored anywhere in your bookmarks (Tab Stash or no). So you can check to see if the star is a solid blue or just an outline.
2. In Tab Stash itself, there are two modes for displaying open tabs—you can ask it to show all open tabs or only unstashed tabs. If your tab does _not_ appear anywhere in `Unstashed Tabs`, you can be sure it's stashed. If it appears in `Open Tabs` and is stashed, it will have a little grey stash icon on the right side of the tab.

Here's an example of what a stashed tab looks like in `Open Tabs`; note the grey icon on the right for the second tab:

<img width="382" alt="Screenshot 2023-10-05 at 18 04 56" src="https://github.com/josh-berry/tab-stash/assets/1203893/ee8daf8f-5d22-4a2a-83ea-65fa5d915bb6">

If you hover over that icon, it will tell you which folder your tabs are stashed in.

Hope this helps!

**Comment by alphapapa** [comment]
Hi Josh,

1.  Of course, you're right, I should have realized that.  Seems that with my current color scheme the filled-in star has poor contrast, so I'd gotten out of the habit of looking at it.
2. Thanks for explaining and showing that.  I must have overlooked that in the options.  And the subtle stash icon, I overlooked that too.

As usual, you've already thought of pretty much everything.  It's those little details that makes Tab Stash the best solution that I've found.  And I appreciate your patiently explaining these things to me as well.  As a developer, I know what it's like to answer questions that are essentially already answered.  :)

Cheers,
Adam
